Application/Api/Billing/Bookkeeping/ApiBasketRepayment.php

```diff
@@ -0,0 +1,360 @@
+            new Panel('Fibu'.new DangerText('*'), array(
+                new TextField('Basket[FibuAccount]', $Account, 'Konto'),
+                new TextField('Basket[FibuToAccount]', $ToAccount, 'Gegenkonto'),
+            ), Panel::PANEL_TYPE_INFO));
+
+    }
+
+    /**
+     * @param string $Identifier
+     * @param string $Type
+     * @param string $BasketId
+     * @param array  $Basket
+     *
+     * @return bool|Well
+     */
+    private function checkInputBasket(
+        $Identifier = '',
+        $Type = '',
+        $BasketId = '',
+        $Basket = array()
+    ){
+
+        $Error = false;
+        $form = $this->formBasket($Identifier, $Type);
+        if(isset($Basket['Name']) && empty($Basket['Name'])){
+            $form->setError('Basket[Name]', 'Bitte geben Sie einen Namen der Abrechnung an');
+            $Error = true;
+        } else {
+            if(isset($Basket['Month']) && isset($Basket['Year'])){
+                // Filtern doppelter Namen Mit Zeitangabe (Namen sind mit anderem Datum wiederverwendbar)
+                if (($tblBasket = Basket::useService()->getBasketByName($Basket['Name'], $Basket['Month'],
+                    $Basket['Year']))){
+                    if ($BasketId !== $tblBasket->getId()){
+                        $form->setError('Basket[Name]',
+                            'Bitte geben sie einen noch nicht vergebenen Name für die Abrechnung '.$Basket['Month'].'.'.$Basket['Year'].' an');
+                        $Error = true;
+                    }
+                }
+            }
+        }
+        if(isset($Basket['TargetTime']) && empty($Basket['TargetTime'])){
+            $form->setError('Basket[TargetTime]', 'Bitte geben Sie ein Fälligkeitsdatum an');
+            $Error = true;
+        }
+        //Rechnungsdatum ist nur bei Datev Pflichtfeld
+        $IsDatev = false;
+        if(($tblSetting = Setting::useService()->getSettingByIdentifier(TblSetting::IDENT_IS_DATEV))){
+            $IsDatev = $tblSetting->getValue();
+        }
+        if($IsDatev && isset($Basket['BillTime']) && empty($Basket['BillTime'])){
+            $form->setError('Basket[BillTime]', 'Bitte geben Sie ein Rechnungsdatum an');
+            $Error = true;
+        }
+        if(isset($Basket['Creditor']) && empty($Basket['Creditor'])){
+            $form->setError('Basket[Creditor]', 'Bitte geben Sie einen Gläubiger an');
+            $Error = true;
+        }
+        // Gutschrift mit Selectbox (eine)
+        if($BasketId == '' && $Basket['Item'] === '0'){
+            $form->setError('Basket[Item]', 'Es wird eine Beitragsart benötigt');
+            $Error = true;
+        }
+
+        if($Error){
+            return new Well($form);
+        }
+
+        return $Error;
+    }
+
+    /**
+     * @param string $Identifier
+     * @param string $Type
+     * @param array  $ErrorHelp
+     *
+     * @return string
+     */
+    public function showAddBasket($Identifier = '', $Type = '', $ErrorHelp = array())
```

Comment:
Codacy found an issue: [Incorrect spacing between argument "$Identifier" and equals sign; expected 0 but found 1](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)